qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ClusteringRuleExtension.java

```diff
@@ -45,14 +61,37 @@ public ClusteringRuleExtension(
     super(partitionCount, replicationFactor, clusterSize, brokerConfigurator, gatewayConfigurator);
   }
 
+  @Override
+  public void afterAll(final ExtensionContext extensionContext) throws Exception {
+    teardown();
+  }
+
+  @Override
+  public void beforeAll(final ExtensionContext extensionContext) throws Exception {
+    invokedStatically = true;
+    setup();
+  }
+
   @Override
   public void afterEach(final ExtensionContext context) throws Exception {
-    super.after();
+    if (!invokedStatically) {
+      teardown();
+    }
+  }
+
+  private void teardown() throws IOException {
     FileUtil.deleteFolderIfExists(tempDir);
+    super.after();
```

Comment:
:x: Should delete the folder after shutting down the brokers. So `super.after()` should be called first.